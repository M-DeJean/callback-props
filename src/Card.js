import React from 'react';
import './Card.css';
import './Messages.css';

// make the ReactDOM available, necessary for rendering the component
import ReactDOM from 'react-dom';

// make the App component available
import App from './App';

export default function Card(props) {
  return (
    <div className='Card'>
      <button
        type='button'
      >
        delete
      </button>
      <h3>{props.title}</h3>
      <p>{props.content}</p>
    </div>
  )
}

// make React available


// this is the test case
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Card />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

export default function Messages(props) {
  const badge = props.unread 
    ? <div className="unread_count">{props.unread}</div> 
    : null;
  return (
    <div className="messages">
      {props.name} 
      {badge}
    </div>  
  );
}

//git push test