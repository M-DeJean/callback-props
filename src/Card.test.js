import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import render from 'react-test-renderer';

// make React available


// this is the test case
it('renders without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');
  
    //  render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<Card />, div);

    ReactDOM.render(<Card 
        title = 'Tenth Card'
        content = 'whatever'
    />, div);
  
    // clean up code
    ReactDOM.unmountComponentAtNode(div);
  });


  it('renders as expected', () => {
    const tree = render.create(<Card
        title = 'Tenth Card'
        content = 'whatever'
  />).toJSON()
  expect(tree).toMatchSnapshot()
});