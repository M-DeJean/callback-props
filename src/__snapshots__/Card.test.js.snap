// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Tenth Card
  </h3>
  <p>
    whatever
  </p>
</div>
`;
